frontend/pages/ManageControlsPage/MacOSSetup/cards/BootstrapPackage/components/BootstrapPackageTable/BootstrapPackageTable.tsx

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import { useQuery } from "react-query";
+
+import { IBootstrapPackageAggregate } from "interfaces/mdm";
+import mdmAPI from "services/entities/mdm";
+
+import DataError from "components/DataError";
+import TableContainer from "components/TableContainer";
+import EmptyTable from "components/EmptyTable";
+
+import {
+  generateTableData,
+  generateTableHeaders,
+} from "./BootstrapPackageTableConfig";
+
+const baseClass = "bootstrap-package-table";
+
+interface IBootstrapPackageTableProps {
+  currentTeamId: number;
+}
+
+const DEFAULT_SORT_HEADER = "hosts";
+const DEFAULT_SORT_DIRECTION = "asc";
+
+const BootstrapPackageTable = ({
+  currentTeamId,
+}: IBootstrapPackageTableProps) => {
+  const { data: bootstrapPackageAggregatem, isLoading, isError } = useQuery<
+    IBootstrapPackageAggregate,
+    Error,
+    IBootstrapPackageAggregate
+  >(
+    ["bootstrap-package-summary", currentTeamId],
+    () => mdmAPI.getBootstrapPackageAggregate(currentTeamId),
+    {
+      retry: false,
+      refetchOnWindowFocus: false,
+    }
+  );
+
+  const tableHeaders = generateTableHeaders();
```

Comment:
Since the table headers from the function never change for this component, I like to export the array directly from the table config rather than wrapping it in a function. I think it is ok as it is here since this is a pretty lightweight table (compared to others like hosts and software) and the function is basically just returning a reference to `defaultTableHeaders` (and it is defined outside of the component so I think that the reference won't change). But I think we have some other spots in the code where we return an array literal or define the array inside the generate function and I believe that would be a new object reference each time the function executes. So maybe that is something we could clean up and implement patterns to help future us navigate this tricky part of the table render cycle.

In components where table headers can change, I like to memoize the `generateTableHeaders` function call to cut down on the re-renders of the full table container.

Reply:
ah good point. I think in this case I will simply return headers from the config file.